lib/src/screens/auth/auth_screen.dart

```diff
@@ -51,80 +55,46 @@ class _AuthScreenState extends State<AuthScreen> {
-                          response.data['tokens']
-                        );
-
-                        //конвертирование  полей  user в  созщданную модель
-                        UserModel userInfo = UserModel.fromJson(
-                          response.data['user']
-                        );
-
-                        //запись userInfo в localStorage
-                        tokensBox.put('email', userInfo.email);
-                        tokensBox.put('nickname', userInfo.nickname);
-
-                        //запись токена в localStorage
-                        tokensBox.put('access', tokensModel.access);
-                        tokensBox.put('refresh', tokensModel.refresh);
-
-                        print(tokensBox.get('access'));
-                        print(tokensBox.get('refresh'));
-                        Navigator.pushReplacementNamed(context, MainRoute);
-
-                      } on DioError catch(e) {
-                        print(e.response!.data);
-                        showCupertinoModalPopup(
-                            context: context,
-                            builder: (context) {
-                              return CupertinoAlertDialog(
-                                title: Text('Ошибка'),
-                                content: Text('Неправильный  логин или пароль!'),
-                                actions: [
-                                  CupertinoButton(
-                                      child: Text('ОК'),
-                                      onPressed: () => Navigator.pop(context)
-                                  )
-                                ],
-                              );
-                            }
-                        );
-                        throw e;
-                      }
-                    },
+                child: BlocConsumer<LogInBloc, LogInState>(
+                  listener: (context, state) {
+                    if (state is LogInLoaded) {
+                      Navigator.pushReplacementNamed(context, MainRoute);
+                    } else if (state is LogInFailed) {
+                      showCupertinoModalPopup(
+                          context: context,
+                          builder: (context) {
+                            return CupertinoAlertDialog(
+                              title: Text('Ошибка'),
+                              content: Text(state.message ?? ''),
+                              actions: [
+                                CupertinoButton(
+                                    child: Text('ОК'),
+                                    onPressed: () => Navigator.pop(context))
+                              ],
+                            );
+                          });
+                    }
+                  },
+                  builder: (context, state) {
+                    return CupertinoBtn(
+                        label: 'Войти',
+                        onPressed: state is LogInLoading ? () {} : () {
```

Comment:
`onPressed` не должен позволять нажать на кнопку, когда происходит авторизация.
Для этого во время загрузки замените пустую функцию `() {}` на `null`.
```suggestion
                        onPressed: state is LogInLoading ? null : () {
```

P.S. если возникает ошибка, что `onPressed` не может быть `null`, то сделайте его `nullable`.